pkg/manager-nvme/manager.go

```diff
@@ -217,8 +217,9 @@ func findStoragePool(storageId, storagePoolId string) (*Storage, *interface{}) {
 	return nil, nil
 }
 
+// CleanupVolumes - remove all volumes other than the list of providingVolumes
 func CleanupVolumes(providingVolumes []ProvidingVolume) {
-	for _, storage := range mgr.storage {
+	for _, storage := range GetStorage() {
```

Comment:
I don't see the benefit of GetStorage().  It's not copying the mgr.storage array. Instead, it's creating a new []*Storage array and filling it with pointers to the individual Storage objects in mgr.storage.
Well, I guess it's preserving any individual Storage objects, so they can't be lost if something else is removing them from mgr.storage...

Reply:
I think the original code was making a copy of each storage object, such that when the namespaces were deleted an the storage was amount was added to the `unallocatedBytes` member of the `Storage` object, that was not reflected in the `mgr.storage` objects themselves.